### ISSUE TITLE
Verify that we can run without manage rights

### ISSUE THREAD

**Issue by yvesgoeleven** [Particular/NServiceBus.AzureServiceBus#528]
At first sight, while reviewing custom check code for https://github.com/Particular/NServiceBus.AzureServiceBus/issues/524, we discovered there is no check to not fail if queue creation is off and we have a connectionstring without manage rights.

Verify that this scenario works as expected.

**Comment by danielmarbach** [comment]
TLDR: We can't

Both endpoint-oriented and forwarding topology run the manage rights pre-startup check which blows up the endpoint when you don't have manage rights even though you never actually specified that you want to run the installers.

Steps to reproduce:

- Create new namespace on ASB
- Create new policy without manage rights
- Execute for example `When_sending_a_base_command` with the root access key once, this then creates all the required queues for this test and the bundles
- Execute the same test again under new policy without manage rights and commented out `EnableInstallers()` in `DefaultServer`

The test will blow up with the following exception:

```
Test failed: System.AggregateException: One or more errors occurred. ---> System.Exception: Pre start-up check failed: Configured to create topology, but have no manage rights for the following namespace(s): `default`.
   at NServiceBus.Sending.PrepareForSending.<OnStart>d__1.MoveNext() in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Transports\Sending.cs:line 38
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.Features.FeatureActivator.<StartFeatures>d__5.MoveNext() in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Features\FeatureActivator.cs:line 71
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.StartableEndpoint.<StartFeatures>d__2.MoveNext() in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\StartableEndpoint.cs:line 59
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.StartableEndpoint.<Start>d__1.MoveNext() in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\StartableEndpoint.cs:line 45
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.AcceptanceTesting.Support.EndpointRunner.<Start>d__12.MoveNext() in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.AcceptanceTesting\Support\EndpointRunner.cs:line 90
```

Thoughts

- Manage rights check should probably not be executed when installers are not configured to run (quickly hacked this into the code and then the test passes)
- SubscriptionManager calls into topology creator which then tries to create queues, topics and subscriptions. Part of the creation process checks whether the queue, topic or subscription already exists. This part of the code throws an `UnauthorizedAccessException`. If we want to support connections without manage rights we would need to bypass the topology creation as well and assume that the customer setup everything correctly up-front. This could mean in the worst case message loss.

Following proposal to move forward:

- Raise issue to disable manage rights check when installer are not configured to run
- Raise an issue to disable topology creation when no manage rights are given and assume everything is correctly setup

@Particular/azure-maintainers thoughts?

**Comment by SeanFeldman** [comment]
> Raise issue to disable manage rights check when installer are not configured to run

👍 

> Raise an issue to disable topology creation when no manage rights are given and assume everything is correctly setup

👍 

I would also suggest to move Manage Rights check out of the pre-startup check phase as it's clearly a wrong place for it to be invoked. It should be invoked _before_ infrastructure is attempted to be created, assuming installers are turned on.

**Comment by danielmarbach** [comment]
Raised #545 and #546 Closing

**Comment by SeanFeldman** [comment]
#545 & #546 both target `hotfix-7.2.3` but I see if this was also [ported to `develop`](https://github.com/Particular/NServiceBus.AzureServiceBus/commits/develop). @danielmarbach ?

**Comment by danielmarbach** [comment]
Will star the legwork today

**Comment by danielmarbach** [comment]
Smoke tested 7.2.3 by disabling installers and start the samples without the debugger attached. PR https://github.com/Particular/docs.particular.net/pull/2791

**Comment by danielmarbach** [comment]
7.2.3 is now on the staging environment.

**Comment by danielmarbach** [comment]
7.1.8 and 7.2.3 released. All done
